File: src/services/Catalogo/Catalogo.API/Data/Dto/CatalogoDto.cs
namespace Catalogo.API.Data.Dto
{
  public record CatalogoDto
  {
    public string ProdutoId { get; set; } = null!;
    public string Nome { get; set; } = null!;
    public string Descricao { get; set; } = null!;
    public string UnidadeMedida { get; set; } = null!;
    public string ImageUrl { get; set; } = null!;
    public int Estoque { get; set; }
    public decimal Preco { get; set; }
    public double Rating { get; set; }
    public int RatingCount { get; set; }
    public bool IsAtivo { get; set; }
    public bool IsFavorito { get; set; } = false;
  }
}

File: src/mobile/lib/app/utils/dto/catalogo/catalogo_dto.dart
class CatalogoDto {
  CatalogoDto({
    required this.id,
    required this.nome,
    required this.imageUrl,
    required this.preco,
    required this.unidadeMedida,
    required this.estoque,
    required this.isAtivo,
  });
  late final String id;
  late final String nome;
  late final String imageUrl;
  late final double preco;
  late final String unidadeMedida;
  late final int estoque;
  late final bool isAtivo;

  CatalogoDto.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    nome = json['nome'];
    imageUrl = json['imageUrl'];
    preco = json['preco'];
    unidadeMedida = json['unidadeMedida'];
    estoque = json['estoque'];
    isAtivo = json['isAtivo'];
  }

  String getDisponiveis() => estoque == 0
      ? 'Fora de estoque'
      : estoque == 1
          ? '1 disponível'
          : '$estoque disponíveis';
}
